cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretsManagerContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+import java.util.Map;
+
+/**
+ * Secrets Manager context.
+ */
+public interface SecretsManagerContext {
+
+  /**
+   * System properties are derived from the CDAP configuration. Anything in the CDAP configuration that is prefixed by
+   * 'securestore.system.properties.[securestore-name].' will be adding as an entry in the system properties.
+   *
+   * @return unmodifiable system properties for the secrets manager
+   */
+  Map<String, String> getProperties();
+
+  /**
+   * Secrets metadata store to store metadata and associated
+   *
+   * @param dataType data type of the metadata stored
+   * @return secrets metadata store
+   */
+  <T> SecretsMetadataStore<T> getSecretsMetadataStore(Class<T> dataType);
```

Comment:
Is it just for Metadata?

Reply:
What are the possible dataType here? What's the purpose of it?

Reply:
it is for metadata + encrypted secret. The `SecretsMetadataStore` is basically a key value store with key being (namespace, secretName). The dataType is the type of the object stored in the store.

Reply:
If the type is determined by the implementation, then the interface shouldn't expose it as generic type `T`

Reply:
fixed